mavros/src/plugins/global_position.cpp

```diff
@@ -168,12 +170,17 @@ class GlobalPositionPlugin : public plugin::PluginBase {
 		float eph = (raw_gps.eph != UINT16_MAX) ? raw_gps.eph / 1E2F : NAN;
 		float epv = (raw_gps.epv != UINT16_MAX) ? raw_gps.epv / 1E2F : NAN;
 
-		if (!std::isnan(eph)) {
-			const double hdop = eph;
-
-			// From nmea_navsat_driver
-			fix->position_covariance[0 + 0] = fix->position_covariance[3 + 1] = std::pow(hdop, 2);
-			fix->position_covariance[6 + 2] = std::pow(2 * hdop, 2);
+		// With FCU protocol v2.0 use accuracies reported by sensor
+		if (UAS_FCU(m_uas)->get_protocol_version() == mavconn::Protocol::V20 &&
+				raw_gps.h_acc > 0 && raw_gps.v_acc > 0) {
```

Comment:
That is ok, but i think that bit faster: `msg->magic == MAVLINK_STX`.

- `MAVLINK_STX` = 0xfd (v2.0)
- `MAVLINK_STX_MAVLINK1` = 0xfe (v1.0)

Reply:
Also that check is not depend on link configuration, because it always can receive v1 and v2 msgs.

Reply:
Hmm, perhaps it is good idea to add `is_mavlink_v10` and `is_mavlink_v20()` inline helpers...
Will add that after merge.